src/axios-cached-dns-resolve.ts

```diff
@@ -70,13 +69,15 @@ let cachePruneId: NodeJS.Timeout;
 
 init();
 
-export function init() {
+export function init(customConfig?: Config, fresh?: boolean) {
+  // Apply the custom configuration or fallback to the existing config
+  config = customConfig ?? config;
   log = initLogger(config.logging);
 
-  if (config.cache) return;
```

Comment:
We want to force the cache to use the custom config - so had to remove this early return